src/main/java/org/xbill/DNS/CookieOption.java

```diff
@@ -0,0 +1,111 @@
+package org.xbill.DNS;
+
+import java.io.IOException;
+import org.xbill.DNS.utils.base16;
+
+/**
+ * Cookie EDNS0 Option, as defined in https://tools.ietf.org/html/rfc7873
+ *
+ * @see OPTRecord
+ * @author Klaus Malorny
+ */
+public class CookieOption extends EDNSOption {
+
+  /** client cookie */
+  private byte[] clientCookie;
+
+  /** server cookie, may be {@code null} */
+  private byte[] serverCookie;
+
+  /** Default constructor for constructing instance from binary representation. */
+  CookieOption() {
+    super(EDNSOption.Code.COOKIE);
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param clientCookie the client cookie, which must consist of eight bytes
+   */
+  public CookieOption(byte[] clientCookie) {
+    this(clientCookie, null);
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param clientCookie the client cookie, which must consist of eight bytes
+   * @param serverCookie the server cookie, which either must be {@code null} or must consist of 8
+   *     to 32 bytes
+   */
+  public CookieOption(byte[] clientCookie, byte[] serverCookie) {
+    this();
+    if (clientCookie == null) throw new IllegalArgumentException("cookie must not be null");
+    if (clientCookie.length != 8)
+      throw new IllegalArgumentException("cookie must consist of eight bytes");
```

Comment:
`"client cookie...`

Reply:
Sorry, I do not understand; what is exactly the problem with it (and "server cookie", too)?

Reply:
Please add `client` and `server` to the exception messages, otherwise it isn't clear to which parameter the exception refers to.

Reply:
Ok, that makes sense. I didn't notice that.